### PR TITLE
kubevirt: move and refactor request logic from web-ui-components

### PR DESCRIPTION
**Pull request by suomiy**
- replace SafeVM with VMWrapper and MutableVMWrapper
- introduce VMTemplateWrapper

@mareklibra @rawagner Can you take a look? It has changed quite a bit from web-ui-components request.

Import VMWare is still missing - will be done in a followup

### CODE REVIEW

**Comment by suomiy** [comment]
\+ ran the tests and fixed few things

File: frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  getTemplatesWithLabels,
+} from '../../../../selectors/vm-template/advanced';
+import {
+  getFlavorLabel,
+  getOsLabel,
+  getWorkloadLabel,
+} from '../../../../selectors/vm-template/combined-dependent';
+import { TEMPLATE_TYPE_BASE } from '../../../../constants/vm';
+
+export const findTemplate = (
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts]: can we use a `FindTemplateParams` here, somthing liks ?
``` ts
type FindTemplateParams = {
  templates: TemplateKind[],
  {
    userTemplateName: string,
  } | {
    workload: string,
    flavor: string,
    os: string,
 }
}
```

File: frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    os,
+  }: { userTemplateName: string; workload: string; flavor: string; os: string },
+): TemplateKind => {
+  if (userTemplateName) {
+    return templates.find((template) => getName(template) === userTemplateName);
+  }
+  return getTemplatesWithLabels(getTemplatesOfLabelType(templates, TEMPLATE_TYPE_BASE), [
+    getOsLabel(os),
+    getWorkloadLabel(workload),
+    getFlavorLabel(flavor),
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts]: Do we need to warrry about `flavor === 'Cutome'` here ?

File: frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts]: created the type for options

File: frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts (hunk 2 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts]: this is handled by the `getFlavorLabel` function

File: frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts]: just making it a type ... I think that you do not need os, flavor and workload if you have the template ... ?

**Pull request merged**